### PR TITLE
zh: doc for canary upgrade of tidb-operator

### LINKED ISSUE #1044: need canary upgrade tidb operator doc
## Change Request

**This repository is ONLY used to solve issues related to TiDB Operator documentation.
For other issues (related to TiDB, PD, etc), please move to [other repositories](https://github.com/pingcap/).**

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.
#992 

2. Describe your suggestion or addition.

3. Provide some reference materials (documents, websites, etc) if you could.

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)
Fix #1044 

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/canary-upgrade-tidb-operator.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
```
Comment (ran-huang) [inline zh/canary-upgrade-tidb-operator.md:1]: This is not in TOC?

File: zh/canary-upgrade-tidb-operator.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+title: TiDB Operator 灰度升级
+summary: 介绍如何灰度升级 TiDB Operator。
+---
+
+# TiDB Operator 灰度升级
+
+本文介绍如何灰度升级 TiDB Operator。
```
Comment (ran-huang) [inline zh/canary-upgrade-tidb-operator.md:8]: Can you explain what is canary upgrade or what it is used for?

File: zh/canary-upgrade-tidb-operator.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+> **注意：**
+>
+> - 目前仅支持灰度升级 tidb-controller-manager 和 tidb-scheduler。
+> - v1.1.10 开始支持此项功能，所以当前 TiDB Operator 版本要 >= v1.1.10
```
Comment (ran-huang) [inline zh/canary-upgrade-tidb-operator.md:13]:
```suggestion
> - v1.1.10 开始支持此项功能，所以当前 TiDB Operator 版本需 >= v1.1.10。
```

File: zh/canary-upgrade-tidb-operator.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    controllerManager:
+      selector:
+      - version!=canary
+    ```
+
+    如果之前已经执行过上述步骤，可以直接进入下一步。
+
+2. 部署灰度 TiDB Operator。
+
+    参考[部署 TiDB Operator 文档](deploy-tidb-operator.md)，在 values.yaml 中添加如下配置，**在不同的 namespace**，例如，`tidb-admin-canary`，部署灰度 TiDB Operator：
```
Comment (ran-huang) [inline zh/canary-upgrade-tidb-operator.md:35]:
```suggestion
    参考[部署 TiDB Operator 文档](deploy-tidb-operator.md)，在 `values.yaml` 中添加如下配置，**在不同的 namespace** 中（例如 `tidb-admin-canary`）部署灰度 TiDB Operator：
```

File: zh/deploy-multiple-tidb-operator.md [hunk 5]
```diff
@@ -89,7 +90,7 @@ aliases: ['/zh/tidb-in-kubernetes/dev/canary-deployment-tidb-operator/']
 
 3. 部署第二套 TiDB Operator。
 
-    参考[部署 TiDB Operator 文档](deploy-tidb-operator.md)，在 values.yaml 中添加如下配置，部署第二套 TiDB Operator (没有部署 `tidb-scheduler`)：
+    参考[部署 TiDB Operator 文档](deploy-tidb-operator.md)，在 values.yaml 中添加如下配置，**在不同的 namespace**，例如，`tidb-admin-qa`，部署第二套 TiDB Operator (没有部署 `tidb-scheduler`)：
```
Comment (ran-huang) [inline zh/deploy-multiple-tidb-operator.md:93]:
```suggestion
    参考[部署 TiDB Operator 文档](deploy-tidb-operator.md)，在 values.yaml 中添加如下配置，**在不同的 namespace** 中（例如 `tidb-admin-qa`）部署第二套 TiDB Operator (没有部署 `tidb-scheduler`)：
```

File: zh/canary-upgrade-tidb-operator.md (hunk 1 above)
Comment (DanielZhangQD) [reply zh/canary-upgrade-tidb-operator.md:1]: Fixed.

File: zh/canary-upgrade-tidb-operator.md (hunk 4 above)
Comment (DanielZhangQD) [reply zh/canary-upgrade-tidb-operator.md]: Fixed.

**Comment by DanielZhangQD** [comment]
@ran-huang comments addressed, PTAL again.

File: zh/TOC.md [hunk 6]
```diff
@@ -30,6 +30,7 @@
   - [部署 TiDB Binlog](deploy-tidb-binlog.md)
   - [部署 TiDB 企业版集群](deploy-tidb-enterprise-edition.md)
   - [部署多套 TiDB Operator](deploy-multiple-tidb-operator.md)
+  - [灰度升级 TiDB Operator](canary-upgrade-tidb-operator.md)
```
Comment (ran-huang) [inline zh/TOC.md:33]: Do you think it fits more in "运维"?

**Comment by DanielZhangQD** [comment]
@ran-huang comments addressed, PTAL again.

**Review by ran-huang — COMMENTED** [review]
LGTM

**Review by handlerww — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**